crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -389,19 +389,16 @@ def check_repo_tag
     end
 
     def check_key_file
-      expected = @config.fetch("repomd", {})["key_md5"]
+      expected = @config.fetch("repomd", {})["fingerprint"]
       return true if expected.blank?
 
       key_path = repodata_path.join("repomd.xml.key")
       key_path = repodata_media_path.join("repomd.xml.key") unless key_path.file?
 
       if key_path.file?
-        md5 = Digest::MD5.hexdigest(key_path.read)
-        if expected.is_a?(Array)
-          expected.include? md5
-        else
-          md5 == expected
-        end
+        fingerprint = `LC_ALL=C gpg --with-fingerprint #{key_path} | grep fingerprint | awk -F= '{print $2}'`
```

Comment:
>> +        fingerprint = `LC_ALL=C gpg --with-fingerprint #{key_path} | grep fingerprint | awk -F= '{print $2}'`
> 
> just thinking out loud, are we sure that "awk" is always installed?
We probably should try to avoid using it. I'll experiment a bit.